src/main/java/io/druid/data/input/MapBasedRow.java

```diff
@@ -159,6 +161,52 @@ public long getLongMetric(String metric)
   }
 
   @Override
+  public int getIntMetric(String metric)
+  {
+    Object metricValue = event.get(metric);
+
+    if (metricValue == null) {
+      return 0;
+    }
+
+    if (metricValue instanceof Number) {
+      return ((Number) metricValue).intValue();
+    } else if (metricValue instanceof String) {
+      try {
+        return Integer.valueOf(((String) metricValue).replace(",", ""));
```

Comment:
Trivial but how about wrapping this with `if (((String) metricValue).indexOf(',') >=0)`?